From 3b837bb72d6b9ab4d563faa211f90efc257e3c96 Mon Sep 17 00:00:00 2001
From: Reini Urban <rurban@cpan.org>
Date: Tue, 31 Dec 2019 10:19:01 +0100
Subject: [PATCH] free: protect freeing obj->dxfname

dont set INDXF when HANDSEED is missing.
e.g. GH #178 null_pointer2 testcase
---
 programs/dwgrewrite.c | 2 +-
 src/encode.c          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/programs/dwgrewrite.c b/programs/dwgrewrite.c
index 5642addfa..cf40117b1 100644
--- a/programs/dwgrewrite.c
+++ b/programs/dwgrewrite.c
@@ -197,7 +197,7 @@ main (int argc, char *argv[])
   i = optind;
 
   memset (&dwg, 0, sizeof (Dwg_Data));
-  dwg.opts = opts;
+  dwg.opts = opts & 0xf;
 
   filename_in = argv[i];
   if (!filename_in)
diff --git a/src/encode.c b/src/encode.c
index a4d514dcc..84ac4f818 100644
--- a/src/encode.c
+++ b/src/encode.c
@@ -2420,7 +2420,7 @@ dwg_encode_header_variables (Bit_Chain *dat, Bit_Chain *hdl_dat,
 
   if (!_obj->HANDSEED) // minimal or broken DXF
     {
-      dwg->opts |= (DWG_OPTS_MINIMAL | DWG_OPTS_INDXF);
+      dwg->opts |= DWG_OPTS_MINIMAL;
       dat->from_version = dat->version - 1;
       LOG_TRACE ("encode from minimal DXF\n");
       _obj->HANDSEED = calloc(1, sizeof(Dwg_Object_Ref));
